providers/implementations/kdfs/hmacdrbg_kdf.c.in

```diff
@@ -199,7 +199,7 @@ static int hmac_drbg_kdf_set_ctx_params(void *vctx,
     size_t size = 0;
     int md_size;
 
-    if (hmac == NULL || !hmac_drbg_kdf_set_ctx_params_decoder(params, &p))
+    if (!hmac_drbg_kdf_set_ctx_params_decoder(params, &p))
```

Comment:
presumably this NULL check is the same condition as the ones above and below?  Do we need to keep the null check and move the assignment beyond it?